src/Microsoft.ML.Core/EntryPoints/ModuleCatalog.cs

```diff
@@ -40,6 +40,7 @@ public sealed class ModuleCatalog
         /// </summary>
         public sealed class EntryPointInfo
         {
+            public readonly bool NoSeal;
             public readonly string Name;
```

Comment:
Let's revert this change of `NoSeal`. (The other change, the `ObsoleteAttribute` change is good, so please don't revert the whole file.)

Firstly, there's the point as made elsewhere that inheritance wasn't the right way to solve that problem, but there's a larger point...

The `EntryPointAttribute` is a description *of the component*. You wanted that class to be unsealed specifically because you had an idea of how to structure the C# API *specifically*. Setting aside the merits of that idea, or even supposing that it was a good idea, we should avoid putting things that are so extremely specific stuff in one environment (in this case, the code in `CSharpApiGenerator.cs`) in the entry-point definition meant to service everyone. Let's think about everything else in this info class... it's useful and necessary practically everywhere.  This is not the case with `NoSeal`.